app/Models/User.php

```diff
@@ -2,18 +2,19 @@
 
 namespace Gladiator\Models;
 
-use Illuminate\Foundation\Auth\User as Authenticatable;
+use Gladiator\Models\Competition;
+use Gladiator\Models\WaitingRoom;
+use Illuminate\Foundation\Auth\User as BaseUser;
 
-class User extends Authenticatable
+class User extends BaseUser
 {
+
     /**
-     * The attributes that are mass assignable.
+     * Indicates if the IDs are auto-incrementing.
      *
-     * @var array
+     * @var bool
      */
-    protected $fillable = [
-        'first_name', 'last_name', 'email', 'password',
-    ];
+    public $incrementing = false;
```

Comment:
This was the key to making Mongo-like strings work with pivot table assignment work when Laravel typically expects an integer id. When this property was set to `true`, Laravel was casting the Mongo-like strings of user ids to integers... most ended up as `0`, but some would end up as random numbers because the string started with that number, haha (quite confusing at first as to what was going on) Thanks @DFurnes for helping figure this one out!